nodes/src/nodes/dataset_cobalt/dataset_loader.py

```
@@ -0,0 +1,263 @@
+        if '..' in path.replace('\\', '/').split('/'):
+            raise ValueError(f'Path traversal detected in file path: {path}')
+
+        normalized = os.path.normpath(path)
+
+        # Reject absolute paths outside the working directory for security.
+        # Only paths under the current working directory (or relative paths
+        # that resolve into it) are permitted.
+        if os.path.isabs(normalized):
+            cwd = os.getcwd()
+            if not normalized.startswith(cwd):
+                raise ValueError(f'Absolute path not allowed outside working directory: {path}')
```

Ruff (0.15.7)

[warning] Avoid specifying long messages outside the exception class

(TRY003)